crates/bevy_time/src/time.rs

```diff
@@ -210,5 +234,36 @@ mod tests {
             (second_update_instant - start_instant)
         );
         assert_eq!(time.delta_seconds(), time.delta().as_secs_f32());
+        assert_float_eq(
+            time.seconds_since_startup_f32_wrapped(),
+            time.seconds_since_startup() as f32,
+        );
+    }
+
+    #[test]
+    fn update_wrapping() {
+        let start_instant = Instant::now();
+
+        let mut time = Time {
+            startup: start_instant,
+            max_wrapping_period: Duration::from_secs(2),
+            ..Default::default()
+        };
+
+        assert_eq!(time.seconds_since_startup_f32_wrapped(), 0.0);
+
+        time.update_with_instant(start_instant + Duration::from_secs(1));
+        assert_float_eq(time.seconds_since_startup_f32_wrapped(), 1.0);
+
+        time.update_with_instant(start_instant + Duration::from_secs(2));
+        assert_float_eq(time.seconds_since_startup_f32_wrapped(), 2.0);
+
+        // wraps on next update
+        time.update_with_instant(start_instant + Duration::from_secs(3));
+        assert_float_eq(time.seconds_since_startup_f32_wrapped(), 1.0);
+    }
+
+    fn assert_float_eq(a: f32, b: f32) {
+        assert!((a - b) <= f32::EPSILON, "{a} != {b}");
```

Comment:
```suggestion
        assert!((a - b).abs() <= f32::EPSILON, "{a} != {b}");
```
To ensure it will work even with `assert_float_eq(0.1, 0.2)`

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=39bd9f4c370d99ce5ce939a15c88d42d